File: frontend/credit-app/src/app/services/credit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Credit, CreditPersonnel, CreditImmobilier, CreditProfessionnel, StatutCredit } from '../models/credit.model';

@Injectable({
  providedIn: 'root'
})
export class CreditService {
  private apiUrl = 'http://localhost:8080/api/credits';

  constructor(private http: HttpClient) { }

  // General credit methods
  getAllCredits(): Observable<Credit[]> {
    return this.http.get<Credit[]>(this.apiUrl);
  }

  getCreditById(id: number): Observable<Credit> {
    return this.http.get<Credit>(`${this.apiUrl}/${id}`);
  }

  getCreditsByCustomerId(customerId: number): Observable<Credit[]> {
    return this.http.get<Credit[]>(`${this.apiUrl}/customer/${customerId}`);
  }

  getCreditsByStatut(statut: StatutCredit): Observable<Credit[]> {
    return this.http.get<Credit[]>(`${this.apiUrl}/statut/${statut}`);
  }

  // Credit Personnel methods
  createCreditPersonnel(credit: CreditPersonnel): Observable<CreditPersonnel> {
    return this.http.post<CreditPersonnel>(`${this.apiUrl}/personnel`, credit);
  }

  updateCreditPersonnel(id: number, credit: CreditPersonnel): Observable<CreditPersonnel> {
    return this.http.put<CreditPersonnel>(`${this.apiUrl}/personnel/${id}`, credit);
  }

  getCreditPersonnelByMotif(motif: string): Observable<CreditPersonnel[]> {
    return this.http.get<CreditPersonnel[]>(`${this.apiUrl}/personnel/motif/${motif}`);
  }

  // Credit Immobilier methods
  createCreditImmobilier(credit: CreditImmobilier): Observable<CreditImmobilier> {
    return this.http.post<CreditImmobilier>(`${this.apiUrl}/immobilier`, credit);
  }

  updateCreditImmobilier(id: number, credit: CreditImmobilier): Observable<CreditImmobilier> {
    return this.http.put<CreditImmobilier>(`${this.apiUrl}/immobilier/${id}`, credit);
  }

  // Credit Professionnel methods
  createCreditProfessionnel(credit: CreditProfessionnel): Observable<CreditProfessionnel> {
    return this.http.post<CreditProfessionnel>(`${this.apiUrl}/professionnel`, credit);
  }

  updateCreditProfessionnel(id: number, credit: CreditProfessionnel): Observable<CreditProfessionnel> {
    return this.http.put<CreditProfessionnel>(`${this.apiUrl}/professionnel/${id}`, credit);
  }

  getCreditProfessionnelByRaisonSociale(raisonSociale: string): Observable<CreditProfessionnel[]> {
    return this.http.get<CreditProfessionnel[]>(`${this.apiUrl}/professionnel/raison-sociale/${raisonSociale}`);
  }
}

File: backend/src/main/java/com/example/creditapp/services/CreditService.java
package com.example.creditapp.services;

import com.example.creditapp.dtos.CreditDTO;
import com.example.creditapp.dtos.CreditImmobilierDTO;
import com.example.creditapp.dtos.CreditPersonnelDTO;
import com.example.creditapp.dtos.CreditProfessionnelDTO;
import com.example.creditapp.enums.StatutCredit;

import java.util.List;

public interface CreditService {
    List<CreditDTO> getAllCredits();
    CreditDTO getCreditById(Long id);
    List<CreditDTO> getCreditsByCustomerId(Long customerId);
    List<CreditDTO> getCreditsByStatut(StatutCredit statut);
    void deleteCredit(Long id);
    
    CreditPersonnelDTO saveCreditPersonnel(CreditPersonnelDTO creditPersonnelDTO);
    CreditPersonnelDTO updateCreditPersonnel(Long id, CreditPersonnelDTO creditPersonnelDTO);
    List<CreditPersonnelDTO> getCreditPersonnelByMotif(String motif);
    
    CreditImmobilierDTO saveCreditImmobilier(CreditImmobilierDTO creditImmobilierDTO);
    CreditImmobilierDTO updateCreditImmobilier(Long id, CreditImmobilierDTO creditImmobilierDTO);
    
    CreditProfessionnelDTO saveCreditProfessionnel(CreditProfessionnelDTO creditProfessionnelDTO);
    CreditProfessionnelDTO updateCreditProfessionnel(Long id, CreditProfessionnelDTO creditProfessionnelDTO);
    List<CreditProfessionnelDTO> getCreditProfessionnelByRaisonSociale(String raisonSociale);
    
    CreditDTO updateCreditStatus(Long id, StatutCredit statut);
}
